### PR TITLE
Ticket2566 create log file on demand

### PR DESCRIPTION
**Pull request by John-Holt-Tessella**
### Description of work

Create a logging system that logs continually. Not just when the logging ends.

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2566

### Acceptance criteria

see ticket

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Has the author taken into account the multi-threaded nature of the code?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?
 - [ ] Documentation updated in https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Logging-from-the-archive

### Functional Tests

- [ ] Do changes function as described? Add comments below that describe the tests performed.

### Final steps

- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: ArchiverAccess/log_file_generator.py [hunk 1]
```diff
@@ -16,19 +16,12 @@
 """
 Module for creating a log file
```
Comment (ThomasLohnert) [inline ArchiverAccess/log_file_generator.py]: I gather the purpose of this file is for testing large databases - can you please clearly document this somewhere in here?

File: ArchiverAccess/log_file_generator.py [hunk 2]
```diff
@@ -53,18 +46,18 @@ def create_log(headers, columns, time_period, filename_template="default.log", h
         config_builder.table_column(column_header, column_template)
 
     adfc = ArchiveDataFileCreator(config_builder.build(), archiver_data_source)
-    adfc.write(time_period)
+    adfc.write_complete_file(time_period)
 
 if __name__ == '__main__':
```
Comment (ThomasLohnert) [inline ArchiverAccess/log_file_generator.py]: It took 121 seconds to run this file. I am not sure what we consider reasonable so I will just leave this here

**Comment by mattclarke** [comment]
Just to echo Thomas' comments, we did agree a docstring format in a Code Chat, so we should try our best to use it:

File: ArchiverAccess/log_file_generator.py (hunk 1 above)
Comment (John-Holt-Tessella) [reply ArchiverAccess/log_file_generator.py]: Created a script see https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Archive-access

File: ArchiverAccess/time_last_active.py [hunk 3]
```diff
@@ -23,11 +23,14 @@
 from ArchiverAccess.configuration import DEFAULT_LOG_PATH
 from server_common.utilities import print_and_log, SEVERITY
 
-TIME_LAST_ACTIVE_HEADER = "# File containing two line, time the logger was last active and " \
-                          "maximum number of days use the time last active"
+TIME_FORMAT = "%Y-%m-%dT%H:%M:%S"
```
[the remark this answers is not in the document]
Comment (John-Holt-Tessella) [reply ArchiverAccess/time_last_active.py]: This is the standard date-time format see https://en.wikipedia.org/wiki/ISO_8601

**Pull request merged**